src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/LaunchProfilesDebugLaunchProvider.cs

```diff
@@ -113,6 +114,34 @@ private async Task<IReadOnlyList<IDebugLaunchSettings>> QueryDebugTargetsInterna
             return null;
         }
 
+        private class LaunchCompleteCallback : IVsDebuggerLaunchCompletionCallback
+        {
+            private readonly DebugLaunchOptions _launchOptions;
+            private readonly IDebugProfileLaunchTargetsProvider? _targetProfile;
+            private readonly ILaunchProfile? _activeProfile;
+
+            public LaunchCompleteCallback(DebugLaunchOptions launchOptions, IDebugProfileLaunchTargetsProvider? targetProfile, ILaunchProfile? activeProfile)
+            {
+                _launchOptions = launchOptions;
+                _targetProfile = targetProfile;
+                _activeProfile = activeProfile;
+            }
+
+            public void OnComplete(int hr, uint debugTargetCount, VsDebugTargetProcessInfo[] processInfoArray)
+            {
```

Comment:
Is this called on the UI thread or a background thread?

Reply:
Was going to check about that  - this doesn't have a guaranteed thread it'll called back on (though generally it will be either the UI thread or a debug engine thread.

Reply:
Isn't switching to the UI thread the responsibility of OnAfterLaunchAsync since it has an async signature?